flink-ml-lib/src/main/java/org/apache/flink/ml/common/broadcast/BroadcastStreamingRuntimeContext.java

```diff
@@ -0,0 +1,68 @@
+package org.apache.flink.ml.common.broadcast;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.api.common.accumulators.Accumulator;
+import org.apache.flink.api.common.functions.BroadcastVariableInitializer;
+import org.apache.flink.api.common.state.KeyedStateStore;
+import org.apache.flink.metrics.groups.OperatorMetricGroup;
+import org.apache.flink.runtime.execution.Environment;
+import org.apache.flink.runtime.externalresource.ExternalResourceInfoProvider;
+import org.apache.flink.runtime.jobgraph.OperatorID;
+import org.apache.flink.streaming.api.operators.StreamingRuntimeContext;
+import org.apache.flink.streaming.runtime.tasks.ProcessingTimeService;
+
+import javax.annotation.Nullable;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * An subclass of {@link StreamingRuntimeContext} that provides accessibility of broadcast
+ * variables.
+ */
+public class BroadcastStreamingRuntimeContext extends StreamingRuntimeContext {
+
+    Map<String, List<?>> broadcastVariables = new HashMap<>();
+
+    public BroadcastStreamingRuntimeContext(
+            Environment env,
+            Map<String, Accumulator<?, ?>> accumulators,
+            OperatorMetricGroup operatorMetricGroup,
+            OperatorID operatorID,
+            ProcessingTimeService processingTimeService,
+            @Nullable KeyedStateStore keyedStateStore,
+            ExternalResourceInfoProvider externalResourceInfoProvider) {
+        super(
+                env,
+                accumulators,
+                operatorMetricGroup,
+                operatorID,
+                processingTimeService,
+                keyedStateStore,
+                externalResourceInfoProvider);
+    }
+
+    @Override
+    public boolean hasBroadcastVariable(String name) {
+        return broadcastVariables.containsKey(name);
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public <RT> List<RT> getBroadcastVariable(String name) {
+        return (List<RT>) broadcastVariables.get(name);
```

Comment:
Do we need to consider the case that users call `getBroadcastVariable` before received all the elements of broadcastVariables ?

Reply:
Hi Yun, good question.

I would like to throw a exception if users call `getBroadcastVariable` before received all the elements of broadcastVariables, i.e., before calling `processElement` or `processWatermark`. How do you think?